app/_src/policies/meshtimeout.md

```diff
@@ -0,0 +1,315 @@
+---
+title: Mesh Timeout (beta)
+---
+
+{% warning %}
+This policy uses new policy matching algorithm and is in beta state,
+it should not be mixed with [Timeout policy](../timeout).
+{% endwarning %}
+
+## TargetRef support matrix
+
+| TargetRef type    | top level | to  | from |
+| ----------------- | --------- | --- | ---- |
+| Mesh              | ✅         | ✅   | ✅    |
+| MeshSubset        | ✅         | ❌   | ❌    |
+| MeshService       | ✅         | ✅   | ❌    |
+| MeshServiceSubset | ✅         | ❌   | ❌    |
+| MeshGatewayRoute  | ❌         | ❌   | ❌    |
+
+To learn more about the information in this table, see the [matching docs](/docs/{{ page.version }}/policies/targetref).
+
+## Configuration
+
+This policy enables {{site.mesh_product_name}} to set timeouts on the inbound and outbound connections depending on the protocol. Using this policy you can configure TCP and HTTP timeouts. Timeout configuration is split into two sections: common configuration and http configuration. Common config is applied to both HTTP and TCP communication.
+
+MeshTimeout policy lets you configure multiple timeouts:
+
+- connectionTimeout
+- idleTimeout
+- http requestTimeout
+- http streamIdleTimeout
+- http maxStreamDuration
+- http maxConnectionDuration
+
+### Timeouts explained
+
+#### Connection timeout
+
+Connection timeout specifies the amount of time DP will wait for a TCP connection to be established.
+
+#### Idle timeout
+
+Idle timeout is the amount of time that the DP will allow a connection to exist with no inbound or outbound activity.
+
+#### HTTP request timeout
+
+Request timeout lests you configure how long the DP should wait for the full response. In details it spans between the point at which the entire request has been processed by DP and when the response has been completely processed by DP.
+
+#### HTTP stream idle timeout
+
+Stream idle timeout is the amount of time that the DP will allow a stream to exist with no inbound or outbound activity. This timeout is strongly recommended for all requests (not just streaming requests/responses) as it additionally defends against a peer that does not open the stream window once an entire response has been buffered to be sent to a downstream client.
+
+#### HTTP max stream duration
+
+Max stream duration is the maximum time that a stream’s lifetime will span. You can use this functionality when you want to reset HTTP request/response streams periodically.
+
+#### HTTP max connection duration
+
+Max connection duration is the time after which a inbound or outbound connection will be drained and/or closed, starting from when it was first established. If there are no active streams, the connection will be closed. If there are any active streams, the drain sequence will kick-in, and the connection will be force-closed after 5 seconds.
+
+### Examples
+
+#### Simple outbound HTTP configuration
+
+This configuration will be applied to all dataplanes inside of Mesh.
```

Comment:
some time ago we decided that we want to use `data plane proxies` instead of `dataplanes`. You still may see `dataplanes` in docs here and there but we should stick to `data plane proxies`

Reply:
Good to know, do we have written somewhere guidelines what naming we should use in docs?

Reply:
I don't think we do :(

Reply:
https://github.com/kumahq/kuma-website/issues/1217